src/webgl/primitives.js

```diff
@@ -800,59 +800,110 @@ p5.RendererGL.prototype.triangle = function(args) {
+};
 
-      for (var i = 0; i < this.detailX; i++) {
-        var u = i / this.detailX;
-        var theta = 2 * Math.PI * u;
+p5.RendererGL.prototype.arc = function(args){
 
-        var _x = 0.5 + Math.cos(theta) / 2;
-        var _y = 0.5 + Math.sin(theta) / 2;
+   var x       = arguments[0];
+   var y       = arguments[1];
+   var width   = arguments[2];
+   var height  = arguments[3];
+   var start   = arguments[4];
+   var stop    = arguments[5];
+   var mode    = arguments[6];
+   var detailX = arguments[7] || 25;
+   var detailY = 1;
 
-        this.vertices.push(new p5.Vector(_x, _y, 0));
-        this.uvs.push(_x, _y);
+   var gId = 'arc|' + x + "|" + y + "|" + width + "|" + 
+   height + "|" + start + "|" + stop + "|" + mode + "|" + detailX + "|";
+
+   if (!this.geometryInHash(gId))
+   {
+
+    var _arc = function()
+    {
+        this.strokeIndices = [];
+
+        for (var i = 0; i <= this.detailX; i++)
+        {
+            var u = i / this.detailX;
+            var theta = (stop - start) * u + start;
+
+            var _x = 0.5 + Math.cos(theta)/2;
+            var _y = 0.5 + Math.sin(theta)/2;
+
+            this.vertices.push(new p5.Vector(_x, _y, 0));
+            this.uvs.push([_x, _y]);
+
+           if (i < this.detailX && mode == PIE)
+           {                                               
+                this.faces.push([0, i + 1, i + 2]);
+                this.strokeIndices.push([i + 1, i + 2]);
+           }
+
+           if (i < this.detailX - 1 && (mode == CHORD || mode == OPEN || mode == 'ellipse'))
+           {
+                this.faces.push([0, i + 1, i + 2]);
+                this.strokeIndices.push([i + 1, i + 2]);
+           }
+
+        }
+
+        if (mode == PIE || mode == null){
+            this.vertices.unshift(new p5.Vector(0.5,0.5,0));
+            this.uvs.unshift([0.5,0.5,0]);
+            this.faces.push([0, this.vertices.length - 2, this.vertices.length - 1]);
+            this.strokeIndices.push([0 , 1]);
+            this.strokeIndices.push([0, this.vertices.length - 1]);
+        }
+
+        else if (mode == CHORD){
+            this.strokeIndices.push([0 , 1]);
+            this.strokeIndices.push([0, this.vertices.length - 1]);
+        }
+
+        else if (mode == OPEN || mode == 'ellipse') {
+            this.strokeIndices.push([0 , 1]);
+        }
 
-        this.faces.push([0, (i + 1) % this.detailX + 1, i + 1]);
-      }
     };
-    var ellipseGeom = new p5.Geometry(detailX, 1, _ellipse);
-    ellipseGeom.computeNormals();
+
+    var arcGeom = new p5.Geometry(detailX, 1, _arc);
+    arcGeom.computeNormals();
+
     if (detailX <= 50) {
-      ellipseGeom._makeTriangleEdges()._edgesToVertices();
-    } else {
-      console.log('Cannot stroke ellipse with more than 50 detailX');
+      arcGeom._makeTriangleEdges()._edgesToVertices(arcGeom);
+    } 
+    else {
+      console.log('Cannot stroke arc with more than 50 detailX');
```

Comment:
a nice detail here might be to say "ellipse" instead when appropriate; so that user will see the name of the method they called

Reply:
Fixed.